walletlibrarydemo/src/main/java/com/microsoft/walletlibrarydemo/feature/issuance/viewlogic/RequirementsFragment.kt

```diff
@@ -0,0 +1,193 @@
+package com.microsoft.walletlibrarydemo.feature.issuance.viewlogic
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.microsoft.walletlibrary.requests.VerifiedIdIssuanceRequest
+import com.microsoft.walletlibrary.requests.VerifiedIdPresentationRequest
+import com.microsoft.walletlibrary.requests.VerifiedIdRequest
+import com.microsoft.walletlibrary.requests.requirements.GroupRequirement
+import com.microsoft.walletlibrary.requests.requirements.VerifiedIdRequirement
+import com.microsoft.walletlibrary.verifiedid.VerifiableCredential
+import com.microsoft.walletlibrary.verifiedid.VerifiedId
+import com.microsoft.walletlibrary.verifiedid.VerifiedIdClaim
+import com.microsoft.walletlibrarydemo.databinding.RequirementsFragmentBinding
+import com.microsoft.walletlibrarydemo.feature.issuance.presentationlogic.*
+import kotlinx.coroutines.runBlocking
+
+class RequirementsFragment : Fragment(), ClickListener {
+    private var _binding: RequirementsFragmentBinding? = null
+    private val binding get() = _binding!!
+
+    private val viewModel: SampleViewModel by viewModels {
+        ViewModelFactory(requireContext())
+    }
+
+    private val args by navArgs<RequirementsFragmentArgs>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = RequirementsFragmentBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        configureViews()
+    }
+
+    private fun configureViews() {
+        binding.requestCompletion.setOnClickListener { completeRequest() }
+        binding.home.setOnClickListener { goBackHome() }
+        binding.requirementsList.layoutManager = LinearLayoutManager(context)
+        binding.requirementsList.isNestedScrollingEnabled = false
+        runBlocking {
+            viewModel.initiateRequest(args.requestUrl)
```

Comment:
add a comment here to highlight this is the important part.

Reply:
done.